### ISSUE TITLE
Errors on enrollment to a course using enrollment code

### ISSUE THREAD

**Issue by michzimny** [up2university/docker-moodle#80]
When I enrol using code 'v9rmpj' to a course in https://learn-dev.test.up2university.eu/ I get the following error. After clicking "Continue", it seems to works fine, and the course creator role seems to be granted by "Category Auto Enrol" block present in that course.

```

Notice: Undefined variable: category_enrol_config in /var/www/html/blocks/category_enrol/classes/observer.php on line 49

Notice: Trying to get property of non-object in /var/www/html/blocks/category_enrol/classes/observer.php on line 49
Skip to main content
This page should automatically redirect. If nothing is happening please use the continue link below.
(Continue)

Error output, so disabling automatic redirect.

```

**Comment by michzimny** [comment]
It seems it happens even when I enroll, as a non-admin user, to a course that does not have "Category Auto Enrol" block.

**Comment by nadavkav** [comment]
Weird. it used to work. Did you set it up correctly before you tried to enrol into the course?
The error is actually saying "The roles where not defined on the block setting page"

**Comment by nadavkav** [comment]
I will see if I can replicate this on my local development server

**Comment by nadavkav** [comment]
Is anyone using this block?

**Comment by michzimny** [comment]
> Did you set it up correctly before you tried to enrol into the course?

What do you mean? I have created the new category with the new course inside of the category. To the course, I have added the block with settings as in your video: https://www.dropbox.com/s/86vwi277j2r9bbu/capture-block.PNG?dl=0

> Is anyone using this block?

I created this block yesterday and started testing.

**Comment by nadavkav** [comment]
@michzimny sorry, but I do not have time for this this week. only at the end of next week.

**Comment by nadavkav** [comment]
@michzimny 
Sorry for the very very loooong delay in reply to this issue.
I managed to reproduce this on the learn-dev.test server and also on my new local development server,
And also figure it out :smile: , as it was due to a situation where a course had several blocks, which funny enough I did not consider with my initial test in which there was only the auto category enrollment block, and that is why I did not manage to reproduce it on my old/original development server.
Also fixed some other tiny issues and updated the block too.
Long story short... It's is fixed.

Here is a new version: 
https://github.com/iucc/moodle-block_category_enrol
Please update your learn-dev.test Moodle instance and tell me if it works.

**Comment by michzimny** [comment]
I've pushed the update of your plugin: https://github.com/up2university/docker-moodle/commit/1938787718d5fdf08d388d03dd5b6c72fc9fbba1

I've tested, and it seems the issue is fixed.
